### ISSUE TITLE
Performance Alerting should work for individual transactions automatically - Group by in Alert rules

### ISSUE THREAD

**Issue by Dhrumil-Sentry** [getsentry/sentry#66871]
### Problem Statement

Feedback from Discord:

> Somewhat related I have created an alert for p95(transaction.duration) > 300ms. Based on my tests it looks like the p95 is computed across all transactions and not after grouping by transaction.name. Is that correct? If so how can I compute the p95 after grouping? Such a grouping is important because say I have 2 endpoints myApp/XXX and myApp/YYY and 99% of the calls go to myApp/XXX which always returns in 100ms I will not be able to detect if any the myApp/YYY calls take an exceptionally long amount of time.
> 
> Setting the alerts individually will not work as we have 100s of different transaction names

### Product Area

Alerts

**Comment by kerenkhatiwada** [comment]
Hey team, I have another user requesting the same [here](https://sentry.zendesk.com/agent/tickets/169537). Is there an updated timeline for this feature?

**Comment by malwilley** [comment]
This work is still planned, but is dependent on the new detector platform first being released. A conservative estimate would be in the next 6 months.

**Comment by Kobby-Bawuah** [comment]
Hey @malwilley 

Another customer is requesting this. Their use case is Spans Metric Alerts grouped by span.description for JS bundle size tracking. Can you confirm the work behind workflow-engine-metric-alert-group-by-creation will cover Spans/EAP alerts?

Zendesk: https://sentry.zendesk.com/agent/tickets/180213

**Comment by malwilley** [comment]
Yes, I can confirm that group by will cover spans, so you will be able to group by span.description and alert based on that once the feature is released.

**Comment by alicankaralar** [comment]
As i understand it, I need this feature too, also on Issue Alerts

Usecase: Canary deployments on 5 different regions. It'd be great to have only 1 alert that fires an integration which calls an endpoint to automate a rollback to the stable track for a given region. 
Region is a tag (context) value. So i need a group by functionality for tags in alerts.
Since I have a lot of alerts, it's getting really hard to maintain all these alerts and I'd like a feature to be able to fire an alert based on a monitor for a specific region - with the region data in the context when the integration is calling the rollback endpoint so we can automatically roll back the problematic region (aka grouped by region).

Is my assumption correct?

**Comment by malwilley** [comment]
@alicankaralar creating issues grouped by a `region` tag (or any tag) will be supported. You should only need to create a single monitor (e.g. "More than 10 errors in {{region}}") and you will get unique issues created for each region value.

The automation portion should also be supported. The created issues will have tag values like `region:us` and if your integration is able to check for that on the event data, everything _should_ work as you'd like it to.
